### DISCUSSION TITLE
Documentation and sample code

### DISCUSSION THREAD

**Discussion by fmatosqg** [enisdenjo/graphql-sse#16] [Q&A] [answered]
I have interest in implementing a kotlin client (for android, possibly multiplatform) version of graphql subscriptions over SSE, and this project seems to be aligned with what I need.

For context pls see https://github.com/apollographql/apollo-kotlin/issues/3756

However I'm not a javascript developer, and the several snippets of code present in the README.md are not really useful for me, since I'd need to pick one, learn the related framework, THEN start my own piece of work. So I'm looking for any standalone sample projects that implement either client or server, ready to be cloned and run from command line without any coding. Are you aware of something like this? I mean, I'd prefer a project + set of instructions that someone that a less than an enthusiastic jr javascript could handle and still be jr at the end of putting it to run. It doesn't help that I'm not fond of js. 

On the other hand, are you aware of any client implementations of this protocol not directly related to this github repository?

**Comment by enisdenjo** [comment, marked as answer]
Hey, happy to hear you're considering using GraphQL over SSE! Sadly, I am not aware of any boilerplate project nor a custom client.

For implementing your own server/client, you should find [the GraphQL over Server-Sent Events protocol](https://github.com/enisdenjo/graphql-sse/blob/master/PROTOCOL.md) sufficient. For any details you might not understand there, I am more than happy to help.

**Reply by fmatosqg** [reply]
I got myself familiar with that doc, as far as I understand my server team we'll go over the http1 with single connection. 

So far so good :) I'm now writing my own kotlin ktor server (for tests) in parallel with my server team progress.

One part that we're going to need given our infrastructure, is that we will need the `INIT` request reply with a url, that is going to be the long lived SSE connection. That's because how it's going to be deployed in our company. Looking forward to extending your official protocol to make that an optional step. Ideas?

**Reply by enisdenjo** [reply]
> One part that we're going to need given our infrastructure, is that we will need the `INIT` request reply with a url, that is going to be the long lived SSE connection.

I don't quite follow. Can you elaborate a bit more?
